htdocs/core/class/commondocgenerator.class.php

```diff
@@ -496,9 +496,13 @@ public function get_substitutionarray_object($object, $outputlangs, $array_key =
 			// Note that this added fields correspond to nothing in Dolibarr (Dolibarr manage discount on lines not globally)
 			$resarray['object_total_up'] = $totalUp;
 			$resarray['object_total_up_locale'] = price($resarray['object_total_up'], 0, $outputlangs);
-			if (method_exists($object, 'getTotalDiscount')) {
-				$resarray['object_total_discount'] = round(100 / $totalUp * $object->getTotalDiscount(), 2);
+			$totalDiscount=$object->getTotalDiscount();
```

Comment:
Depending on object received in parameter, the method getTotalDiscount may not exists.